bot/utils/parameter_value_generator.py

```diff
@@ -14,27 +23,18 @@ def parameter_value_generator(chemistry, parameter):
     Returns:
         param_value: numerical
     """
+    params = pybamm.ParameterValues(chemistry=chemistry)
+    base_value = {}
 
-    if parameter == "Current function [A]":
-        if chemistry == pybamm.parameter_sets.Chen2020:
-            param_value = desired_decimal_point_generator(3, 5, 2)
-
-        elif chemistry == pybamm.parameter_sets.Marquis2019:
-            param_value = desired_decimal_point_generator(0.1, 0.65, 2)
-
-        elif chemistry == pybamm.parameter_sets.Ai2020:
-            param_value = desired_decimal_point_generator(0.5, 2.25, 2)
-
-        elif chemistry == pybamm.parameter_sets.Yang2017:
-            param_value = desired_decimal_point_generator(0.5, 2.25, 2)
-
-        elif chemistry == pybamm.parameter_sets.Chen2020_plating:
-            param_value = desired_decimal_point_generator(3, 5, 2)
-
+    if callable(params[parameter]):
+        base_value[parameter] = 1.0
+        params[parameter] = FunctionLike(params[parameter], parameter)
     else:
-        params = pybamm.ParameterValues(chemistry=chemistry)
-        param_value = desired_decimal_point_generator(
-            params[parameter]*0.9, params[parameter]*1.1, 2
-        )
+        base_value[parameter] = params[parameter]
+        params[parameter] = pybamm.InputParameter(parameter)
+
+    param_value = desired_decimal_point_generator(
+        (base_value[parameter] - 0.1)*0.5, (base_value[parameter] + 0.1)*2, 2
```

Comment:
What is the reasoning behind these limits? Note that between approx 0.1 and 0.8 the first argument is larger than the second (not sure if that's a problem). Also, in the first argument you get complex values if `base_value < 0.1`.

Reply:
The +-0.1 was done to ensure that if the base_value is 0 then all the generated values don't come out to be 0. I don't remember specifically but there were some parameters values that were set to 0 by default and there I was encountering this problem.

An alternative might be to just set the value of the parameter (where the default value is 0) as 0.1 (before generating the random value) and then remove the +-0.1 thing, would this be a better implementation?

Reply:
Are you just changing the parameters in the list `param_to_vary_list`? I don't think any of these are zero?

Reply:
Oh, weird, I did encounter some cases in the old code, might be some other issue. I'll remove the +-0.1 thing.